components/codec/src/byte.rs

```diff
@@ -918,10 +917,9 @@ mod tests {
                     MemComparableByteCodec::encode_all(src, dest);
                 }
             }
-            #[allow(clippy::needless_range_loop)]
-            for i in encoded_prefix_len + encoded_len..encoded_suffix_len {
-                payload_encoded[i] = rand::random();
-            }
+            payload_encoded[encoded_prefix_len + encoded_len..encoded_suffix_len]
```

Comment:
This line panics when `encoded_suffix_len < encoded_prefix_len + encoded_len`. The original code doesn't panic but it does look strange. Should the range here be `encoded_prefix_len + encoded_len..`?  @breeswish

Reply:
Yes, they are equal, since `payload_encoded`'s length is `encoded_prefix_len + encoded_len + encoded_suffix_len`.

Reply:
But the upper bound of the range in the original code is `encoded_suffix_len`... Isn't it a bug?

Reply:
Yes. Looks like it is a bug..